src/commands/android/subcommands/install/system-image.ts

```diff
@@ -0,0 +1,123 @@
+import colors from 'ansi-colors';
+import inquirer from 'inquirer';
+
+import Logger from '../../../../logger';
+import {APILevelNames} from '../../constants';
+import {Platform} from '../../interfaces';
+import {getBinaryLocation} from '../../utils/common';
+import {execBinarySync, spawnCommandSync} from '../../utils/sdk';
+import {showMissingBinaryHelp} from '../common';
+import {AvailableSystemImages} from '../interfaces';
+
+export async function installSystemImage(sdkRoot: string, platform: Platform): Promise<boolean> {
+  try {
+    const sdkmanagerLocation = getBinaryLocation(sdkRoot, platform, 'sdkmanager', true);
+    if (!sdkmanagerLocation) {
+      showMissingBinaryHelp('sdkmanager');
+
+      return false;
+    }
+
+    const stdout = execBinarySync(sdkmanagerLocation, 'sdkmanager', platform, '--list | grep "system-images;"');
+    if (!stdout) {
+      Logger.log(`${colors.red('Failed to fetch system images!')} Please try again.`);
+
+      return false;
+    }
+
+    // sdkmanager output has repetitive system image names in different sections (Installed
+    // packages, Available packages, Available updates).
+    // Parse the output and store the system images in a Set to avoid duplicates.
+    const images = new Set<string>();
+    // Before removing duplicates, sort the system images to get them in increasing order of API level.
+    const lines = stdout.split('\n').sort();
+
+    lines.forEach(line => {
+      const image = line.split('|')[0].trim();
+      images.add(image);
+    });
+
+    // System images are represented in the format: system-images;android-<api-level>;<type>;<arch>
+    // Group all the system image types by API level. Group all the architectures by system image type.
+    const availableSystemImages: AvailableSystemImages = {};
+
+    images.forEach(image => {
+      if (!image.includes('system-image')) {
+        return;
+      }
+      const imageSplit = image.split(';');
+      const apiLevel = imageSplit[1];
+      const type = imageSplit[2];
+      const arch = imageSplit[3];
+
+      if (!availableSystemImages[apiLevel]) {
+        availableSystemImages[apiLevel] = [];
+      }
+
+      const imageType = availableSystemImages[apiLevel].find(image => image.type === type);
+      if (!imageType) {
+        availableSystemImages[apiLevel].push({
+          type: type,
+          archs: [arch]
+        });
+      } else {
+        imageType.archs.push(arch);
+      }
+    });
+
+    const apiLevelsWithNames = Object.keys(availableSystemImages).map(apiLevel => {
+      if (APILevelNames[apiLevel]) {
+        return `${apiLevel}: ${APILevelNames[apiLevel]}`;
+      }
+
+      return apiLevel;
+    });
+
+    const androidVersionAnswer = await inquirer.prompt({
+      type: 'list',
+      name: 'androidVersion',
+      message: 'Select the API level for system image:',
+      choices: apiLevelsWithNames
+    });
+    const apiLevel = androidVersionAnswer.androidVersion.split(':')[0];
```

Comment:
We should pass an array of objects to the `choices` parameter of `inquirer` so that we can just specify the text and value separately and don't need to extract the actual value from the selected option again by using `split()` as done is last line above.